src/top-bar/period-select/period-select.js

```diff
@@ -0,0 +1,41 @@
+import i18n from '@dhis2/d2-i18n'
+import React from 'react'
+import { ContextSelect } from '../context-select/index.js'
+import { useSelection } from '../selection/index.js'
+
+const PERIOD = 'PERIOD'
+
+const PeriodSelect = () => {
+    const { period, workflow, selectPeriod, openedSelect, setOpenedSelect } =
+        useSelection()
+    const open = openedSelect === PERIOD
+    const value = period.displayName || i18n.t('Choose a period')
+
+    return (
+        <ContextSelect
+            label={i18n.t('Period')}
+            value={value}
+            open={open}
+            disabled={!workflow.id}
+            onOpen={() => setOpenedSelect(PERIOD)}
+            onClose={() => setOpenedSelect('')}
+            tooltipContent={i18n.t('Choose a workflow first')}
+        >
+            <pre>
+                Period picker placeholder <br />
+                <button
+                    onClick={() =>
+                        selectPeriod({
+                            code: '20210404',
+                            displayName: '04-04-2020',
+                        })
+                    }
```

Comment:
Question: Should this also reset the org unit?

Reply:
I checked with Joe about this:
- Clear all should reset all
- Workflow select should clear period and org-unit
- Changes to period and org-unit should not clear anything
Changes related to that behaviour are going to be included in another PR I have been preparing.